packages/zoe/src/contracts/callSpread/pricedCallSpread.js

```diff
@@ -127,8 +128,9 @@ const start = zcf => {
 
       // assert that the requested option was the right one.
       assert(
-        spreadAmount.Option.value[0].instance ===
-          desiredOption.value[0].instance,
+        getCopyBagEntries(spreadAmount.Option.value)[0][0].instance ===
+          // @ts-expect-error runtime narrow
+          getCopyBagEntries(desiredOption.value)[0][0].instance,
         'wanted option not a match',
       );
```

Comment:
First, stop using the two-arg form of `assert`. Use the `|| Fail...` patterns instead.

```suggestion
        getCopyBagEntries(spreadAmount.Option.value)[0][0].instance ===
          // @ts-expect-error runtime narrow
          getCopyBagEntries(desiredOption.value)[0][0].instance ||
        Fail`wanted option not a match`;
```

Next, this `.value)[0][0]` pattern is really ugly. This change warrants a specialized helper, probably to be reused in lots of places we use invitations

```suggestion
        getInvitationEntry(spreadAmount.Option).instance ===
          // @ts-expect-error runtime narrow
          getInvitationEntry(desiredOption).instance ||
        Fail`wanted option not a match`;
```

where that helper can do additional error checks, to ensure that it's assumptions are correct. Something like

```js
const getInvitationEntry = invitationAmount => {
  const value = amountMath.getValue(invitationBrand, invitationAmount);
  const entries = getCopyBagEntries(value);
  entries.length === 1 || Fail`...`;
  entries[0][1] === 1n;
  return entries[0][0];
};
harden(getInvitationEntry);
```

Reply:
> This change warrants a specialized helper

agreed.

> stop using the two-arg form of assert.

The `assert` pre-dates this PR. Let's keep the scope bounded.

Reply:
Bounding the scope that way is fine. Not bounding this case is also fine, since you're touching that `assert` anyway ;)